include/seqan/stream/stream_base.h

```diff
@@ -206,7 +206,7 @@ guessFormatFromStream(TStream &istream, Tag<TFormat_>)
 
     // check magic header
     unsigned i;
-    for (i = 0; i != sizeof(MagicHeader<TFormat>::VALUE) / sizeof(char); ++i)
+    for (i = 0; i != sizeof(MagicHeader<TFormat>::VALUE) / (sizeof(char)); ++i)
```

Comment:
[FIX] suppress Wsizeof-pointer-div

Hä? What?

Reply:
There is a warning when you do this kind of division. It's supposed to warn if you try to get the size of an array by dividing the sizeof the array by the sizeof a pointer to the array. You can silence it with parentheses...

Reply:
Interesting, do you know why this is in general bad, why this is okay here?

Reply:
It's only a problem if the array is passed as function parameter (which we are not doing here).

from the patch to gcc (https://patchwork.ozlabs.org/project/gcc/patch/AM4PR0701MB2162D22EC7A12EB0797E231DE41B0@AM4PR0701MB2162.eurprd07.prod.outlook.com/):
```cpp
int
f1 (int *array)
{
  int i;
  i = sizeof array / sizeof *array;		/* { dg-warning "dividing the pointer size by the element size" } */
  i += sizeof array / sizeof array[0];		/* { dg-warning "dividing the pointer size by the element size" } */
  i += sizeof(array) / sizeof(*array);		/* { dg-warning "dividing the pointer size by the element size" } */
  i += sizeof(array) / sizeof(array[0]);	/* { dg-warning "dividing the pointer size by the element size" } */
  i += (sizeof(array)) / (sizeof(array[0]));	/* { dg-warning "dividing the pointer size by the element size" } */
  i += sizeof(array) / sizeof(int);		/* { dg-warning "dividing the pointer size by the element size" } */
  i += sizeof(array) / sizeof(char);
  i += sizeof(*array) / sizeof(char);
  i += sizeof(array[0]) / sizeof(char);
  return i;
}

int
f2 (void)
{
  int array[10];
  int i;
  i = sizeof array / sizeof *array;
  i += sizeof array / sizeof array[0];
  i += sizeof(array) / sizeof(*array);
  i += sizeof(array) / sizeof(array[0]);
  i += (sizeof(array)) / (sizeof(array[0]));
  i += sizeof(array) / sizeof(int);
  i += sizeof(array) / sizeof(char);
  i += sizeof(*array) / sizeof(char);
  i += sizeof(array[0]) / sizeof(char);
  return i;
}
```